### ISSUE TITLE
update-dbainstance cannot install latest security update for SQL 2019 cu 16

### ISSUE THREAD

**Issue by jensenjk** [dataplat/dbatools#8473]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

WARNING: [13:19:44][Get-SqlInstanceUpdate] 15.0.4223 on computer [P-SQL-SQLE34.Neas.local] is already the latest available.

DBA tools seems to have the buildlevel set for the origunal CU16 - the level for the security update is 
PS C:\WINDOWS\system32> Get-DbaBuildReference -Kb KB5014353

Build          : 15.0.4236
BuildLevel     : 15.0.4236
CULevel        : CU16
KBLevel        : 5014353
MatchType      : Exact
NameLevel      : 2019
SPLevel        : RTM
SupportedUntil : 01/07/2025 00:00:00
Warning        : 

Checking the server gets the following:Build          : 15.0.4223
BuildLevel     : 15.0.4223
BuildTarget    : 15.0.4236
Compliant      : False
CULevel        : CU16
CUTarget       : 
KBLevel        : 5011644
MatchType      : Exact
MaxBehind      : 
NameLevel      : 2019
SPLevel        : RTM
SPTarget       : 
SqlInstance    : P-SQL-SQLE34
SupportedUntil : 01/07/2025 00:00:00
Warning        : 

### Steps to Reproduce

```powershell
# provide your command(s) executed pertaining to dbatools
# please include variable values (redacted or fake if needed) for reference
```

### Please confirm that you are running the most recent version of dbatools

Using latest 1.1.121

### What PowerShell host was used when producing this error

Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version

5.1.20348.643

### SQL Server Edition and Build number

15.0.4223.1

### .NET Framework Version

 4.8

**Comment by potatoqualitee** [comment]
Thank you for the report. Can you please run it with -Verbose and let us know the outcome?

**Comment by jensenjk** [comment]
With verbose
PS C:\WINDOWS\system32>   Update-DbaInstance -ComputerName $computer -Restart -Path $sqlcufilepath -Confirm:$true -credential $pscredential -ExtractPath "C:\ito" -verbose
VERBOSE: [13:58:00][Resolve-DbaNetworkName] Resolving p-sql-sqle34 using .NET.Dns GetHostEntry
VERBOSE: [13:58:00][Resolve-DbaNetworkName] Resolving 10.228.18.221 using .NET.Dns GetHostByAddress
VERBOSE: [13:58:00][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [13:58:00][Get-DbaCmObject] [p-sql-sqle34] Retrieving Management Information
VERBOSE: [13:58:00][Get-DbaCmObject] [p-sql-sqle34] Accessing computer using Cim over WinRM
VERBOSE: [13:58:00][Get-DbaCmObject] [p-sql-sqle34] Accessing computer using Cim over WinRM - Success
VERBOSE: [13:58:01][Resolve-DbaNetworkName] Resolving P-SQL-SQLE34.Neas.local using .NET.Dns GetHostEntry
VERBOSE: [13:58:01][Resolve-DbaBuild] Looking for 15.0.4223
VERBOSE: [13:58:02][Resolve-DbaBuild] We have 26 builds in store for this Release
VERBOSE: [13:58:02][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [13:58:02][Get-DbaCmObject] [p-sql-sqle34.neas.local] Retrieving Management Information
VERBOSE: [13:58:02][Get-DbaCmObject] [p-sql-sqle34.neas.local] Accessing computer using Cim over WinRM
VERBOSE: [13:58:02][Get-DbaCmObject] [p-sql-sqle34.neas.local] Accessing computer using Cim over WinRM - Success
VERBOSE: [13:58:02][Test-PendingReboot] Reboot pending in the PendingFileRenameOperations registry value
VERBOSE: [13:58:02][Update-DbaInstance] Attempting to test Credssp protocol for remote connections
VERBOSE: [13:58:02][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [13:58:02][Get-DbaCmObject] [p-sql-sqle34.neas.local] Retrieving Management Information
VERBOSE: [13:58:02][Get-DbaCmObject] [p-sql-sqle34.neas.local] Accessing computer using Cim over WinRM
VERBOSE: [13:58:02][Get-DbaCmObject] [p-sql-sqle34.neas.local] Accessing computer using Cim over WinRM - Success
VERBOSE: [13:58:02][Resolve-DbaBuild] Looking for 15.0.4223
VERBOSE: [13:58:02][Resolve-DbaBuild] We have 26 builds in store for this Release
WARNING: [13:58:03][Get-SqlInstanceUpdate] 15.0.4223 on computer [P-SQL-SQLE34.Neas.local] is already the latest available.
VERBOSE: [13:58:03][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [13:58:03][Get-DbaCmObject] [p-sql-sqle34.neas.local] Retrieving Management Information
VERBOSE: [13:58:03][Get-DbaCmObject] [p-sql-sqle34.neas.local] Accessing computer using Cim over WinRM
VERBOSE: [13:58:03][Get-DbaCmObject] [p-sql-sqle34.neas.local] Accessing computer using Cim over WinRM - Success
VERBOSE: [13:58:03][Resolve-DbaBuild] Looking for 15.0.4223
VERBOSE: [13:58:03][Resolve-DbaBuild] We have 26 builds in store for this Release
WARNING: [13:58:03][Get-SqlInstanceUpdate] 15.0.4223 on computer [P-SQL-SQLE34.Neas.local] is already the latest available.

**Comment by niphlod** [comment]
I *think* it's because Update-DbaInstance targets "cu level" patches by design if not explicitely forced to.

Try with the `-KB ` param, see [Example 4](https://docs.dbatools.io/Update-DbaInstance)

**Comment by jensenjk** [comment]
using -kb switch works.

It seems there is a mismatch between the 2 functions Test-dbabuild - which finds the update needed - and update-dbainstance which dont update unless you specify KB

**Comment by niphlod** [comment]
in layman terms, usually you want to install "without worries" CUs, not KBs, so Update-DbaInstance in this case has a "sane default", which is also true for Test-DbaBuild, which users usually use to see if there is *something* new to install, regardless of the "CU status"

**Comment by jensenjk** [comment]
We have 2012, 2016, 2017 and 2019 servers.
We just whant to update them in one run - if using KB switch we need split it into 4 different stream which is not optimal.

**Comment by niphlod** [comment]
and if your policy is "install the latest CU" you're perfectly fine. this is just an instance that doesn't happen often where a KB is pushed after a CU.
